### DISCUSSION TITLE
Should I use skip_children or not?

### DISCUSSION THREAD

**Discussion by ProfDoof** [rust-lang/rustfmt#6345] [Q&A] [answered]
Hi all! 

I'm trying to fix rustfmt in treefmt-nix to meet their standards with regards to formatters. They require that a formatter only format a single file per file passed. The formatter shouldn't format any of the out of line modules. 

However, I have been doing some digging into options to ensure that rustfmt does so and that led me to discover the skip_children option. Then, much to my chagrin I discovered that it had been deprecated! Or so I thought. However, turns out that it was only deprecated on the v2 branch and not the v1 branch. Since it was never deprecated on the v1 branch it does still exist. However, it's unclear when it should be used or even how it should be used. If I want to only format the file I requested formatted, and I don't want any file formatted that hasn't been requested, what should I do? 

Do I use skip_children? If so, how? 
Do I suck it up and deal with it? If so, are there any workarounds?
Do I need to use a specific version of rust alongside it? I see something with regards to nightly in the code but I didn't have the time to dig into that any deeper. 

Thanks for the answers y'all!

**Comment by calebcartwright** [comment, marked as answer]
If you're on nightly you can certainly use `skip_children`, and that's used commonly in situations where people/tools only want to format a single file.

You can fully ignore the 2.0 branch, there's not going to be a new major release of rustfmt (outside the entire Rust toolchain having a 2.0 release).

We may one day do a soft-deprecation of `skip_children` (e.g. instead of `skip_children` with a default of `false` we could have something like `format_children` with a default of `true` to avoid the double negative, but we also maintain automatic mappings for these types of cases)
